contracts/src/Gateway.sol

```diff
@@ -596,4 +581,8 @@ contract Gateway is IGatewayBase, IGatewayV1, IGatewayV2, IInitializable, IUpgra
     function initialize(bytes calldata data) external virtual {
         Initializer.initialize(data);
     }
+
+    function createAgent(bytes32 id) external {
```

Comment:
Two issues:

1. First the naming scheme for Gateway V2 public methods is `v2_createAgent`.
2. You'll notice that functions in `Gateway.sol` are grouped together, and in a specific order. The best place to define this function is below `v2_registerToken`. That's where all the public methods for V2 are defined.